azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -224,6 +218,226 @@ def create(
+                original_endpoint.connection_string = get_servicebus_topic_cstring(
+                    cmd=self.cli,
+                    namespace_name=endpoint_account_name,
+                    topic_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+        elif endpoint_type in [
+            EndpointType.EventHub.value, EndpointType.ServiceBusQueue.value, EndpointType.ServiceBusTopic.value
+        ] and entity_path and not connection_string:
+            # only set entity_path if no connection string
+            original_endpoint.entity_path = entity_path
+
+        if endpoint_type == EndpointType.AzureStorageContainer.value:
+            if fetch_connection_string:
+                original_endpoint.connection_string = get_storage_cstring(
+                    cmd=self.cli,
+                    account_name=endpoint_account_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            if container_name:
+                original_endpoint.container_name = container_name
+            if encoding:
+                original_endpoint.encoding = encoding.lower()
+            if file_name_format:
+                original_endpoint.file_name_format = file_name_format
+            if batch_frequency:
+                original_endpoint.batch_frequency = batch_frequency
+            if chunk_size_window:
+                original_endpoint.chunk_size_window = (chunk_size_window * BYTES_PER_MEGABYTE)
+        elif endpoint_type == EndpointType.CosmosDBContainer.value:
+            if fetch_connection_string:
+                # try to get connection string - this will be used to get keys + uri
+                if not primary_key and not secondary_key:
+                    connection_string = get_cosmos_db_cstring(
+                        cmd=self.cli,
+                        account_name=endpoint_account_name,
+                        rg=endpoint_resource_group,
+                        sub=endpoint_subscription_id
+                    )
+            if connection_string:
+                # parse out key from connection string
+                parsed_cs = parse_cosmos_db_connection_string(connection_string)
+                if not primary_key and not secondary_key:
+                    original_endpoint.primary_key = parsed_cs["AccountKey"]
+                    original_endpoint.secondary_key = parsed_cs["AccountKey"]
+                # parse out endpoint uri from connection string
+                if not endpoint_uri:
+                    original_endpoint.endpoint_uri = parsed_cs["AccountEndpoint"]
+            if database_name:
+                original_endpoint.database_name = database_name
+            if container_name:
+                original_endpoint.container_name = container_name
+            if partition_key_name:
```

Comment:
Not sure how much the service team will care if we allow the user to null out the partition key template (and keep the partition key name). For create, they want us to set it to a default in this scenario

Reply:
We can always ask the service folks for their considerations, but in the past we've been committed to letting the user input values as they see fit, stopping them if there's a conflict (like mutually exclusive arguments, invalid values, etc) and letting the service tell us if something went wrong processing their input.

Reply:
ok will leave it as power to the user